### PR TITLE
Added mean and variance for NoncentralBeta

### PR DESCRIPTION
**Pull request by ArunS-tack**
Working on #1698, I discovered that NoncentraBeta Distribution does not have mean and variance functions. Opening a PR to tackle that.

### CODE REVIEW

**Review by ararslan — COMMENTED** [review]
Thanks for the contribution! Just some comments and questions.

File: src/Distributions.jl [hunk 1]
```diff
@@ -8,6 +8,9 @@ import Base: size, length, convert, show, getindex, rand, vec, inv
 import Base: sum, maximum, minimum, extrema, +, -, *, ==
 import Base.Math: @horner
 
+using HypergeometricFunctions
+import HypergeometricFunctions: pFq
```
[the remark this answers is not in the document]
Comment (ArunSanganal) [reply src/Distributions.jl:12]: Oh, alright. It's my first time adding a dependency to a package, I will keep that in mind.

File: src/univariate/continuous/noncentralbeta.jl [hunk 2]
```diff
@@ -35,11 +35,16 @@ partype(::NoncentralBeta{T}) where {T} = T
 
 ### Evaluation & Sampling
 
-# TODO: add mean and var
+mean(d::NoncentralBeta) = (d.α / (d.α + d.β)) * exp(-d.λ / 2) * HypergeometricFunctions.pFq([1 + d.α, d.α + d.β],[d.α, d.α + d.β + 1], d.λ / 2)
```
[the remark this answers is not in the document]
Comment (ArunSanganal) [reply src/univariate/continuous/noncentralbeta.jl:38]: We could but I actually followed [this](https://reference.wolfram.com/language/ref/NoncentralBetaDistribution.html) assuming they give the same result.

**Comment by ArunSanganal** [comment]
Thank you for the review. I have committed the suggestions and will take care in future to make the code as clean as possible.

**Comment by ArunSanganal** [comment]
Oops. Sorry for so many review requests, I thought the it was bugged and clicked it way too many times.😓

File: test/univariate/continuous/noncentralbeta.jl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using Distributions
+using Test
+
+
+@testset "NoncentralBeta" begin 
+    α = 2.0; β = 3.0; λ = 1.0
```
Comment (devmotion) [inline test/univariate/continuous/noncentralbeta.jl:6]: Could we actually test more random/less nice values?  Or directly use the R reference setup?

File: src/univariate/continuous/noncentralbeta.jl [hunk 4]
```diff
@@ -35,11 +35,22 @@ partype(::NoncentralBeta{T}) where {T} = T
 
 ### Evaluation & Sampling
 
-# TODO: add mean and var
+function mean(d::NoncentralBeta)
+    α, β, λ = params(d)
+    return α / (α + β) * exp(-λ / 2) * pFq([1 + α, α + β], [α, α + β + 1], λ / 2)
```
Comment (devmotion) [inline src/univariate/continuous/noncentralbeta.jl:40]: I wonder if this can be simplified further, given e.g. the formula in https://en.wikipedia.org/wiki/Generalized_hypergeometric_function#Generalization_of_Kummer's_transformations_and_identities_for_2F2. It seems, however, that the first identity leads to incorrect results here since one would end up with just `alpha / (alpha  + beta)` - I guess there is some assumption on a, b, c, and d that is not satisfied here.

(Not our fault but also a bit annoying that one has to call `pFq` with vectors instead of tuples)

Comment (ArunSanganal) [reply src/univariate/continuous/noncentralbeta.jl:40]: [Wolfram](https://reference.wolfram.com/language/ref/NoncentralBetaDistribution.html) is the reference I used. Can you think of any edge case where the exception might be triggered?

File: test/univariate/continuous/noncentralbeta.jl [hunk 5]
```diff
@@ -0,0 +1,15 @@
+using Distributions
+using Test
+
+
+@testset "NoncentralBeta" begin 
+    α = 11.0 ; β = 8.0; λ = 5.0
```
Comment (devmotion) [inline test/univariate/continuous/noncentralbeta.jl:6]: Could we use non-integers?
